src/AspNetCore.VersionInfo/Extensions/VersionInfoBuilderExtensions.cs

```diff
@@ -1,12 +1,36 @@
-﻿using AspNetCore.VersionInfo.Providers;
+﻿using System;
+using System.Collections.Generic;
+using AspNetCore.VersionInfo.Configuration;
+using AspNetCore.VersionInfo.Providers;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace AspNetCore.VersionInfo
 {
     public static class VersionInfoBuilderExtensions
     {
-        public static VersionInfoBuilder With<T>(this VersionInfoBuilder builder) where T : class, IInfoProvider
+        private static List<String> keyValues = new List<String>();
+        public static VersionInfoBuilder With<T>(this VersionInfoBuilder builder, Action<ExclusionSettings> configureOptions = null) where T : class, IInfoProvider
```

Comment:
In this way configureOptions is specific for the provider configuration, but all (excluded) keys of each provider are merged in a single array (static).

Reply:
I think a better approach is using AspNet Options framework... as example you could inspect AspnetCore.Healthchecks code for it

https://github.com/dotnet/aspnetcore/tree/main/src/HealthChecks/HealthChecks/src/DependencyInjection